dashboard/modules/reporter/reporter_agent.py

```diff
@@ -371,6 +374,28 @@ async def CpuProfiling(self, request, context):
         )
         return reporter_pb2.CpuProfilingReply(output=output, success=success)
 
+    async def MemoryProfiling(self, request, context):
+        pid = request.pid
+        format = request.format
+        leaks = request.leaks
+        duration = request.duration
+        native = request.native
+        trace_python_allocators = request.trace_python_allocators
+        p = MemoryProfilingManager(self._log_dir)
+        success, profiler_filename, output = await p.attach_profiler(
+            pid, native=native, trace_python_allocators=trace_python_allocators
+        )
+        if not success:
+            return reporter_pb2.MemoryProfilingReply(output=output, success=success)
+
+        # add 1 second sleep for memray overhead
+        await asyncio.sleep(duration + 1)
+        success, output = await p.detach_profiler(pid)
+        success, output = await p.get_profile_result(
+            pid, profiler_filename=profiler_filename, format=format, leaks=leaks
+        )
+        return reporter_pb2.MemoryProfilingReply(output=output, success=success)
```

Comment:
so `success` and `output` is overwritten by `get_profile_result`? This seems weird. If we don't intend to use the output of `detach_profile` we should simply do `await p.detach_profiler(pid)`

Reply:
and is `detach` necessary as a seperate API from `get_prfoile_result` in this case?

Reply:
Good catch. For this one, it's not affecting `get_profile_result` since memray will generate the bin file when it's attached and `get_profile_result` can give result without requiring memray to be detached.
It might be good to throw a warning here, I'll see how difficult to do that